internal/backends/postgresql/metadata/registry.go

```diff
@@ -701,6 +713,256 @@ func (r *Registry) CollectionRename(ctx context.Context, dbName, oldCollectionNa
 	return true, nil
 }
 
+// IndexesCreate creates indexes in the collection.
+//
+// Existing indexes with given names are ignored.
+//
+// If the user is not authenticated, it returns error.
+func (r *Registry) IndexesCreate(ctx context.Context, dbName, collectionName string, indexes []IndexInfo) error {
+	defer observability.FuncCall(ctx)()
+
+	p, err := r.getPool(ctx)
+	if err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	r.rw.Lock()
+	defer r.rw.Unlock()
+
+	return r.indexesCreate(ctx, p, dbName, collectionName, indexes)
+}
+
+// indexesCreate creates indexes in the collection.
+//
+// Existing indexes with given names are ignored.
+//
+// It does not hold the lock.
+func (r *Registry) indexesCreate(ctx context.Context, p *pgxpool.Pool, dbName, collectionName string, indexes []IndexInfo) error {
+	defer observability.FuncCall(ctx)()
+
+	_, err := r.collectionCreate(ctx, p, dbName, collectionName)
+	if err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	db := r.colls[dbName]
+	if db == nil {
+		panic("database does not exist")
+	}
+
+	c := r.collectionGet(dbName, collectionName)
+	if c == nil {
+		panic("collection does not exist")
+	}
+
+	allIndexes := make(map[string]string, len(db))   // to check if the index already exists
+	allPgIndexes := make(map[string]string, len(db)) // to ensure there are no indexes with the same name in the pg schema
+
+	for _, coll := range db {
+		for _, index := range coll.Indexes {
+			allIndexes[index.Name] = coll.Name
+			allPgIndexes[index.PgIndex] = coll.Name
+		}
+	}
+
+	created := make([]string, 0, len(indexes))
+
+	for _, index := range indexes {
+		if coll, ok := allIndexes[index.Name]; ok && coll == collectionName {
+			continue
+		}
+
+		tableNamePart := c.TableName
+		tableNamePartMax := maxIndexNameLength/2 - 1 // 1 for the separator between table name and index name
+
+		if len(tableNamePart) > tableNamePartMax {
+			tableNamePart = tableNamePart[:tableNamePartMax]
+		}
+
+		indexNamePart := specialCharacters.ReplaceAllString(strings.ToLower(index.Name), "_")
+
+		h := fnv.New32a()
+		must.NotFail(h.Write([]byte(collectionName)))
+		s := h.Sum32()
+
+		var pgIndexName string
+
+		for {
+			suffixHash := fmt.Sprintf("_%08x_idx", s)
+			if l := maxIndexNameLength/2 - len(suffixHash); len(indexNamePart) > l {
+				indexNamePart = indexNamePart[:l]
+			}
```

Comment:
The logic can be simplifies if we concat `tableNamePart` + `_` + `indexNamePart` first then apply logic about cutting index name and adding revolving hash :thinking:

I guess current approach will leave index name even if table name is very long, which makes index identification easier?

Reply:
Yes, I wanted this approach on purpose as I wanted to have at least a tiny part of the index name even if the table name is long.